### PR TITLE
feat(lang/py): add trust for pip_index

### LINKED ISSUE #1387: feat: should support --trusted-host while using `pip_index`
## Description

Sometimes, we may deploy our own pypi mirror, however, there seems not be a function for setting private `pip_index` as `trusted-host`

### PR DESCRIPTION
**Pull request by weixiao-huang**
fix #1387

### CODE REVIEW

**Review by gaocegege — APPROVED** [review]
Thanks for your contribution! :tada: :+1:

/cc @kemingy 

I think we should add it in v1, too.

File: pkg/lang/frontend/starlark/v0/config/config.go [hunk 1]
```diff
@@ -95,18 +95,19 @@ func ruleFuncJupyter(thread *starlark.Thread, _ *starlark.Builtin,
 func ruleFuncPyPIIndex(thread *starlark.Thread, _ *starlark.Builtin,
 	args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
 	var url, extraURL starlark.String
+	var trust starlark.Bool
```
Comment (kemingy) [inline pkg/lang/frontend/starlark/v0/config/config.go:98]: You can use golang bool directly.

File: pkg/lang/frontend/starlark/v0/config/config.go [hunk 2]
```diff
@@ -95,18 +95,19 @@ func ruleFuncJupyter(thread *starlark.Thread, _ *starlark.Builtin,
...
 
 	if err := starlark.UnpackArgs(rulePyPIIndex, args, kwargs,
-		"url?", &url, "extra_url?", &extraURL); err != nil {
+		"url?", &url, "extra_url?", &extraURL, "trust", &trust); err != nil {
```
Comment (kemingy) [inline pkg/lang/frontend/starlark/v0/config/config.go:101]: This should be optional.
```suggestion
		"url?", &url, "extra_url?", &extraURL, "trust?", &trust); err != nil {
```

File: pkg/lang/frontend/starlark/v0/config/config.go (hunk 1 above)
Comment (weixiao-huang) [reply pkg/lang/frontend/starlark/v0/config/config.go]: done

File: pkg/lang/frontend/starlark/v0/config/config.go (hunk 2 above)
Comment (weixiao-huang) [reply pkg/lang/frontend/starlark/v0/config/config.go]: done

**Review by terrytangyuan — APPROVED** [review]

**Comment by gaocegege** [comment]
@all-contributors Please add @weixiao-huang for the code contribution!

Thanks for your contribution! :tada: :+1:

**Pull request merged**